File: framework/test2/common/intersect/intersect.pl
use strict;
use warnings;

# step [1]: implement solution function
# @param {number[]} nums1
# @param {number[]} nums2
# @return {number[]}
sub intersection {
    my ($nums1, $nums2) = @_;
    if (@{$nums1} > @{$nums2}) {
        return intersection($nums2, $nums1);
    }
    my %mset;
    my %res;
    for my $n (@{$nums1}) {
        $mset{$n} = 1;
    }
    for my $b (@{$nums2}) {
        $res{$b} = 1 if exists $mset{$b};
    }
    [map { int $_ } keys %res];
}

use Soda::Unittest::Utils qw(from_stdin);
use Soda::Unittest::TestWork qw(create);
use Soda::Unittest::Validators qw(for_array for_array2d);
# step [2]: setup function/return/arguments
my $work = create(\&intersection);
$work->validator(for_array('number', 0));
$work->compare_serial(1);
print $work->run(from_stdin());

File: framework/test2/common/intersect/intersect.php
<?php
require_once 'Soda/DS/index.php';
require_once 'Soda/Unittest/index.php';

use Soda\DS;
use Soda\Unittest\TestWork;
use Soda\Unittest\Utils;
use Soda\Unittest\Validators;

class Solution {

    /**
     * @param Integer[] $nums1
     * @param Integer[] $nums2
     * @return Integer[]
     */
    function intersection($nums1, $nums2) {
        if (count($nums1) > count($nums2)) {
            return $this->intersection($nums2, $nums1);
        }
        $mset = array();
        $res = array();
        foreach ($nums1 as $n) {
            $mset[$n] = true;
        }
        foreach ($nums2 as $b) {
            if (array_key_exists($b, $mset)) {
                $res[$b] = true;
            }
        }
        return array_keys($res);
    }
}

$work = TestWork::create(new Solution(), 'intersection', $argv[0]);
$work->validator = Validators::forArray('Integer', false);
$work->compareSerial = true;
echo $work->run(Utils::fromStdin());
